Develop

Cipa opened the pull request.

Cipa committed:
make sure transalias.class.php works on php 5.2

Cipa committed:
fx for #429

Cipa committed:
fix #403 +

Cipa committed:
add a few more settings that might be needed by custom tvs

Cipa committed:
removed a comment

Cipa committed:
grammar and , fix

Cipa committed:
fix transalias for php >5.3

Cipa committed:
translates with php version check

Cipa committed:
more tables to use datatables

Cipa committed:
removed a line by mistake

Cipa committed:
add database name for backup

TimGS commented:
At a first glance looks all OK, though I have to admit now being a little puzzled as to the state of play with TransAlias. @Cipa - can you summarise the change? 

More generally, I am wondering whether we should still be bothering supporting PHP 5.2. It is rather old now.

-- Tim.

Cipa commented:
@TimGS 
Initially I fixed the code that was making the 5.2 compiler to generate an error because of the anonymous function used on the >5.2 branch
Then Jako suggested a better approach that doesn't check for a php version. I am fine with both solutions but in general I think a solution that doesn't check for php version is better
I have a lot of projects on 5.2(I am not the owner of the server so and I can't upgrade), I recently upgraded some of their projects to latest Clipper so now I can do some stuff for Clipper as well because the only issue I encountered is the trasalis bug and that one is fixed

It would be ashame not to support 5.2 if it's easily doable :)

Cipa committed:
fix #364

Cipa committed:
fix #444

Cipa committed:
fix #446

Cipa committed:
fix #447

Cipa committed:
fixes for #448 and #397

Cipa committed:
fix #314

Cipa committed:
improved solution for #397

If upload is disabled so should any access to the files and dirs,
basically a user can only use the resources(files, images etc) but
modify them in any way

Cipa committed:
fix #456

TimGS commented:
Sorry, haven't yet merged this.  Do we still need it?
